### PR TITLE
[KED-2655] Removed namespace from datasets

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

Datasets in modular pipeline had the entire namespace visible as the front-end name which wasn't user-friendly hence we decided to remove the namespace and just show the node name (the namespace can be figured through the tree navigation) 

For example - 
Before implementation -- the name displayed for the given node was as below 

image

After implementation -- the name displayed for the given node will be 

Screenshot 2021-06-09 at 17 30 46

## Development notes

The ruleset for front-end naming is to remove any dots, underscores or hyphens in the name and capitalize all the words. 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by studioswong** [comment]
Thanks for the work @rashidakanchwala. Using the example in the below image, what is the expected outcome for the name after implementing the work from this PR? ( my guess is that it would be `Set timezones`, with capitalisation for the first letter?) Are there any particular ruleset applied for the naming implemented in this PR?

For future reference, the PR description would be a good place to elaborate the expected outcome / requirements
image

I will review this once I wrap up my current dev work today :)

**Comment by rashidakanchwala** [comment]
> Thanks for the work @rashidakanchwala. Using the example in the below image, what is the expected outcome for the name after implementing the work from this PR? ( my guess is that it would be `Set timezones`, with capitalisation for the first letter?) Are there any particular ruleset applied for the naming implemented in this PR?
> 
> For future reference, the PR description would be a good place to elaborate the expected outcome / requirements
> image
> 
> I will review this once I wrap up my current dev work today :)

hey, I have updated the PR description. Let me know if you are ok with the ruleset or if you want it done differently. As in just capitalizing the first word.

**Comment by studioswong** [comment]
> hey, I have updated the PR description. Let me know if you are ok with the ruleset or if you want it done differently. As in just capitalizing the first word.
> Also, could you please let me know if this is only required for data nodes, because i see this issue in parameters and tasks as well. Thanks

This ruleset applies to all nodes (@yetudada  can confirm). From what it looks it seems that the namespace issue only occurs for the dataset nodes so it might be fine for now, however please check the code for the parameters and task node naming to ensure it does gets applied.

File: package/tests/test_models/test_graph/test_graph_nodes.py
Comment (studioswong) [inline package/tests/test_models/test_graph/test_graph_nodes.py:30]: minor comment: I wonder why are these type of changes introduced?

**Review by studioswong — APPROVED** [review]
I have tested this on my side and works for me, hence approving this on my side.

However, given that this involves the new BE which I am not familiar with, I strongly recommend for @limdauto to review and approve this before merging this in.

File: package/kedro_viz/models/graph.py
Comment (studioswong) [inline package/kedro_viz/models/graph.py:53]: I am no python developer, but I noticed that the namespace issue does not apply to task and parameter nodes, which might suggest that this has been implemented already somewhere in the code which suggests that this might not be needed?

File: package/kedro_viz/models/graph.py
Comment (studioswong) [inline package/kedro_viz/models/graph.py:207]: I noticed task node uses `_pretty_name(getattr(node, "short_name", node.name))` to apply the ruleset in their naming to eliminate the namespaces, I wonder that, instead of setting a new class, perhaps we can reuse that?

**Review by limdauto — APPROVED** [review]
LGTM ⭐ If you merge main into this branch, it should pass.

File: package/kedro_viz/models/graph.py
Comment (limdauto) [reply package/kedro_viz/models/graph.py:53]: The namespace is added explicitly to dataset's name on the Kedro side. It doesn't get added to node's. 

@rashidakanchwala have you checked parameters?

**Review by limdauto — APPROVED** [review]
This LGTM.

**Pull request merged**
